client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -438,41 +460,42 @@ function mediaButton( editor ) {
 		editor.nodeChanged();
 	}
 
-	function toggleSizingControls( event ) {
-		if ( ! event.element || 'IMG' !== event.element.nodeName ) {
+	function toggleSizingControls( increase, event ) {
+		const selectedSite = getSelectedSiteFromState();
+		if ( ! event.element || 'IMG' !== event.element.nodeName || ! selectedSite ) {
 			return;
 		}
 
-		const parsed = MediaSerialization.deserialize( event.element );
+		const parsed = deserialize( event.element );
+		const media = assign( { width: Infinity, height: Infinity }, MediaStore.get( selectedSite.ID, parsed.media.ID ) );
 
-		// Disable sizing toggles if the image is transient
-		let isHidden = parsed.media.transient;
+		// Hide sizing toggles if the image is transient
+		const isHidden = !! parsed.media.transient;
+		this.classes.toggle( 'hidden', isHidden );
 
-		// Disable sizing toggles if the image is of an unknown size
-		if ( ! isHidden ) {
-			isHidden = ! parsed.appearance.size || ! includes( SIZE_ORDER, parsed.appearance.size );
+		// Disable decrease button when smaller than the smallest thumbnail size
+		// and the full size is bigger than the current size
+		if ( ! increase ) {
+			const thumb = MediaUtils.getThumbnailSizeDimensions( SIZE_ORDER[ 0 ], selectedSite );
+			const isDisabled =
+				( parsed.media.width || Infinity ) <= media.width &&
+				( parsed.media.width || Infinity ) <= thumb.width;
+			this.disabled( isDisabled );
 		}
 
-		// Disable sizing toggles if the image is smaller than the smallest
-		// thumbnail size for the site
-		if ( ! isHidden ) {
-			const thumb = MediaUtils.getThumbnailSizeDimensions( SIZE_ORDER[ 0 ], sites.getSelectedSite() );
-			isHidden = ( parsed.media.width || Infinity ) < thumb.width && ( parsed.media.height || Infinity ) < thumb.height;
+		// Disable increase button when we select full-size
+		if ( increase ) {
```

Comment:
Minor: Could this be an `if` `else` instead of two separate `if`s? If you do decide to change, I'd recommend putting the truthy condition first (tends to be easier to read).